pkg/webhook/configmap.go

```diff
@@ -79,20 +130,43 @@ func (mw *MutatingWebhook) MutateConfigMap(configMap *corev1.ConfigMap, vaultCon
 	return nil
 }
 
-func (mw *MutatingWebhook) mutateConfigMapBinaryData(configMap *corev1.ConfigMap, data map[string]string, secretInjector *injector.SecretInjector) error {
-	mapData, err := secretInjector.GetDataFromVault(data)
+// ======== BAO ========
+
+func (mw *MutatingWebhook) mutateConfigMapForBao(configMap *corev1.ConfigMap, baoConfig bao.Config) error {
+	baoClient, err := mw.newBaoClient(baoConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to create bao client")
+	}
+	defer baoClient.Close()
+
+	config := baoinjector.Config{
+		TransitKeyID:     baoConfig.TransitKeyID,
+		TransitPath:      baoConfig.TransitPath,
+		TransitBatchSize: baoConfig.TransitBatchSize,
+	}
```

Comment:
for example, providers could add implementation such as:
```go
// Defines mutator interface that all providers should implement (Bao and Vault)
type Mutator interface {
	MutateConfigMap(...)
	MutateObject(...)
	// other mutators
}

```
this way we move the provider-specific logic to the right place (to avoid mixing the business logic of mutation with provider internals)